openstack/networking/v2/subnets/requests.go

```diff
@@ -112,15 +121,24 @@ type UpdateOptsBuilder interface {
 // UpdateOpts represents the attributes used when updating an existing subnet.
 type UpdateOpts struct {
 	Name           string      `json:"name,omitempty"`
-	GatewayIP      string      `json:"gateway_ip,omitempty"`
+	GatewayIP      *string     `json:"gateway_ip,omitempty"`
```

Comment:
OK, so no changes needed to the Update part then?

If not, I have an extra unit test to push and then I think it's ready for another full review.